### PR TITLE
Cache go mod download

### PR DESCRIPTION
**Pull request by gdbelvin**

### CODE REVIEW

File: cmd/keytransparency-server/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,13 @@
 FROM golang:1.12 as build
 
 WORKDIR /go/src/github.com/google/keytransparency
```
Comment (AlCutter) [inline cmd/keytransparency-server/Dockerfile]: Would be neat if these all shared the same `build` stage, is that possible?

Comment (gdbelvin) [reply cmd/keytransparency-server/Dockerfile]: How do I do that?

Comment (gdbelvin) [reply cmd/keytransparency-server/Dockerfile]: Or what would be the value in doing that?

File: cmd/keytransparency-monitor/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,13 @@
 FROM golang:1.12 as build
 
 WORKDIR /go/src/github.com/google/keytransparency
-COPY . .
+COPY go.mod .
+COPY go.sum .
```
Comment (RJPercival) [inline cmd/keytransparency-monitor/Dockerfile]: Can combine these two steps as `COPY go.mod go.sum ./`

File: cmd/keytransparency-server/Dockerfile (hunk 1 above)
Comment (AlCutter) [reply cmd/keytransparency-server/Dockerfile]: I was thinking that if there were something like:

```
FROM ... as base
...
go mod download
COPY ...
```

which was then shared by all the job-specific Dockerfiles, e.g.:
```

FROM base as build:
go get .../${thing}

FROM distroless/...
COPY --from=build ...
```
 Then you'd save having to rebuild the same `base` image/re-fetching the same modules etc. repeatedly.

**Pull request merged**